lib/browser/guest-window-manager.js

```diff
@@ -267,53 +271,42 @@ ipcMainInternal.on('ELECTRON_GUEST_WINDOW_MANAGER_INTERNAL_WINDOW_OPEN', functio
   }
 })
 
-ipcMainInternal.on('ELECTRON_GUEST_WINDOW_MANAGER_WINDOW_CLOSE', function (event, guestId) {
-  const guestContents = webContents.fromId(guestId)
-  if (guestContents == null) return
-
-  if (!canAccessWindow(event.sender, guestContents)) {
-    console.error(`Blocked ${event.sender.getURL()} from closing its opener.`)
-    return
-  }
+const handleMessage = function (channel, handler) {
+  ipcMainUtils.handle(channel, (event, guestId, ...args) => {
+    const guestContents = webContents.fromId(guestId)
+    if (!guestContents) {
+      throw new Error(`Invalid guestId: ${guestId}`)
+    }
 
-  const guestWindow = getGuestWindow(guestContents)
-  if (guestWindow != null) guestWindow.destroy()
-})
+    return handler(event, guestContents, ...args)
+  })
+}
 
 const windowMethods = new Set([
+  'destroy',
   'focus',
   'blur'
 ])
 
-ipcMainInternal.on('ELECTRON_GUEST_WINDOW_MANAGER_WINDOW_METHOD', function (event, guestId, method, ...args) {
-  const guestContents = webContents.fromId(guestId)
-  if (guestContents == null) {
-    event.returnValue = null
-    return
+handleMessage('ELECTRON_GUEST_WINDOW_MANAGER_WINDOW_METHOD', (event, guestContents, method, ...args) => {
+  if (!canAccessWindow(event.sender, guestContents)) {
+    console.error(`Blocked ${event.sender.getURL()} from calling ${method}`)
+    throw new Error(`Access denied to guestId: ${guestContents.id}`)
   }
 
-  if (!canAccessWindow(event.sender, guestContents) || !windowMethods.has(method)) {
-    console.error(`Blocked ${event.sender.getURL()} from calling ${method} on its opener.`)
-    event.returnValue = null
-    return
+  if (!windowMethods.has(method)) {
+    console.error(`Blocked ${event.sender.getURL()} from calling ${method}`)
```

Comment:
Should this be different from the message up there?

Reply:
what do you mean exactly?

Reply:
I mean that you throw exceptions with two different errors, but log two exactly the same messages to the console.
Maybe the message should be different as well?